src/configuration.ts

```diff
@@ -86,8 +91,9 @@ export class RLSConfiguration {
         this.wsPath = wsPath;
     }
 
-    public rustupConfig(): RustupConfig {
-        return new RustupConfig(this.channel, this.rustupPath);
+    // Added ignoreChannel for readChannel function. Otherwise we end in an infinite loop.
```

Comment:
I guess it'd be better to restructure this but we can leave this for a follow up PR